src/main/java/at/hannibal2/skyhanni/features/garden/pests/PesthunterProfit.kt

```diff
@@ -0,0 +1,193 @@
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalName
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.ItemUtils.itemName
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NumberUtil.shortFormat
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object PesthunterProfit {
+
+    private val config get() = GardenAPI.config.pests.pesthunterShop
+    private var display = emptyList<Renderable>()
+
+    var inInventory = false
+
+    /**
+     * REGEX-TEST: §2100 Pests
+     */
+    private val pestCostPattern by patternGroup.pattern(
+        "garden.pests.inventory.cost",
+        "§2(?<pests>\\d+) Pests"
+    )
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        inInventory = false
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!config.pesthunterProfitEnabled) return
+        if (event.inventoryName != "Pesthunter's Wares") return
+
+        inInventory = true
+
+        val table = mutableListOf<DisplayTableEntry>()
+        for ((slot, item) in event.inventoryItems) {
+            try {
+                readItem(slot, item, table)
+            } catch (e: Throwable) {
+                ErrorManager.logErrorWithData(
+                    e, "Error in PesthunterProfit while reading item '${item.itemName}'",
+                    "item" to item,
+                    "name" to item.itemName,
+                    "inventory name" to InventoryUtils.openInventoryName(),
+                )
+            }
+        }
+
+        val newList = mutableListOf<Renderable>()
+        newList.add(Renderable.string("§ePesthunter Shop Profit"))
+        newList.add(LorenzUtils.fillTable(table, padding = 5, itemScale = 0.7))
+        display = newList
+    }
+
+    private fun readItem(slot: Int, item: ItemStack, table: MutableList<DisplayTableEntry>) {
+        val itemName = item.displayName
+        if (itemName == " ") return
+        if (itemName == "§cClose") return
+        if (itemName == "§aSell Item") return
+        if (itemName == "§6Pesthunter's Wares") return
+
+        val totalCost = getFullCost( getRequiredItems(item) )
+        if (totalCost < 0) return
+
+        val (name, amount) = ItemUtils.readItemAmount(itemName) ?: return
+
+        var internalName = NEUInternalName.fromItemNameOrNull(name)
+        if (internalName == null) {
+            internalName = item.getInternalName()
+        }
+
+        if (itemName == "§7[Lvl 100] §6Hedgehog") {
+            internalName = NEUInternalName.fromItemNameOrInternalName("HEDGEHOG;4")
```

Comment:
this workaround should not be needed. investigate what is going on intead

Reply:
i tried multiple times, absolutely no idea why it doesnt have an internalname or itemname in the shop gui

Reply:
> i tried multiple times, absolutely no idea why it doesnt have an internalname or itemname in the shop gui

You can use the skull data for the hedgehog that you get when copying the item to check against the skull ID/texture in the shop